lib/init.coffee

```diff
@@ -11,4 +15,36 @@ module.exports =
         'Available codes: https://pypi.python.org/pypi/pep257#error-codes')
 
   activate: ->
-    console.log 'activate linter-pep257'
+    @subscriptions = new CompositeDisposable
+    @subscriptions.add atom.config.observe 'linter-pep257.executablePath',
+      (executablePath) =>
+        @executablePath = executablePath
+    @subscriptions.add atom.config.observe 'linter-pep257.ignoreCodes',
+      (ignoreCodes) =>
+        @ignoreCodes = ignoreCodes
+
+  deactivate: ->
+    @subscriptions.dispose()
+
+  provideLinter: ->
+    provider =
+      grammarScopes: ['source.python']
+      scope: 'file' # or 'project'
+      lintOnFly: true # must be false for scope: 'project'
+      lint: (textEditor)->
+        filePath = textEditor.getPath()
+        params = [filePath]
+        if @ignoreCodes
+            params.push("--add-ignore=" + @ignoreCodes)
+        return helpers.exec(@executablePath, params, {stream: 'stderr'}).then (stderr) ->
+          output = stderr.split(/\r?\n/)
+          messages = helpers.parse(stderr,
+                                   ":(?<line>\\d+)\\s+(?<message>.+)",
+                                   {filePath: filePath})
+          messages.forEach (msg, i) ->
+            msg.type = 'Warning'
```

Comment:
I would prefer not to avoid using type `Warning`. The reasoning is that this linter is most likely used with other linters, pep257 messages should clobber more important warnings (like those concerning syntax).

The previous version of linter there was type "Info", but that appears to have been removed and is now "trace" :/

I recommend sticking with "Info", the message badges will not be grey instead of styled blue, but they will not be on-par with other warnings.